File: app/src/main/java/com/todobom/opennotescanner/helpers/ScannedDocument.java
package com.todobom.opennotescanner.helpers;

import org.opencv.core.Mat;
import org.opencv.core.Point;
import org.opencv.core.Size;

/**
 * Created by allgood on 05/03/16.
 */
public class ScannedDocument {

    public Mat original;
    public Mat processed;
    public Quadrilateral quadrilateral;
    public Point[] previewPoints;
    public Size previewSize;

    public ScannedDocument(Mat original) {
        this.original = original;
    }

    public Mat getProcessed() {
        return processed;
    }

    public ScannedDocument setProcessed(Mat processed) {
        this.processed = processed;
        return this;
    }

    public void release() {
        if (processed != null) {
            processed.release();
        }
        if (original != null) {
            original.release();
        }

        if (quadrilateral != null && quadrilateral.contour != null) {
            quadrilateral.contour.release();
        }
    }
}

File: app/src/main/java/com/todobom/opennotescanner/helpers/ScannedDocument.kt
package com.todobom.opennotescanner.helpers

import org.opencv.core.Mat
import org.opencv.core.Point
import org.opencv.core.Size

class ScannedDocument(val original: Mat) {
    var processed: Mat? = null
    var quadrilateral: Quadrilateral? = null
    var previewPoints: Array<Point>? = null
    var previewSize: Size? = null

    fun release() {
        processed?.release()
        original.release()
        quadrilateral?.contour?.release()
    }
}